src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/HashAlgorithm.cs

```diff
@@ -92,6 +92,11 @@ public void Dispose()
             GC.SuppressFinalize(this);
         }
 
+        public void Clear() 
+        {
+            (this as IDisposable).Dispose();
```

Comment:
There's a subtle difference between ```this.Dispose()``` and ```(this as IDisposable).Dispose()```... if a derived type re-implements the IDisposable interface explicitly, the latter will end up using it.  I assume that's the desired behavior?  Just verifying.